### DISCUSSION TITLE
🚀 Feature Request: D1 migrations: Possibility to just apply one migration at a time

### DISCUSSION THREAD

**Discussion by houmark** [cloudflare/workers-sdk#13519] [Feature Requests]
### Describe the solution

It would be nice to have an additional non-mandatory parameter to the CLI command where you can apply just one specific migration. For example:

`wrangler d1 migrations apply [database] [number/version of migration]`

I can see that this may be problematic if you skip an older migration which newer migrations depend on, but in the end, this should be up to the user to ensure (and the migration may error). Alternatively, this feature could allow only the oldest pending migration to apply. This would enable the user to incrementally check each migration before continuing with more migrations.

**Comment by lambrospetrou** [comment]
I am closing this feature request since testing individual queries can be done through the normal query execution flow through the REST API or [`wrangler d1 execute` ](https://developers.cloudflare.com/d1/wrangler-commands/#d1-execute).

**Comment by houmark** [comment]
That's okay, I guess. But this has been a while, and I recall it being if you had a particular query that did something problematic, and you wanted to just force run one existing migration.

But it's been so long that I can barely recall the exact use case, and as you mentioned, there are more flexible ways to handle this now.
